Open Judge System/Web/OJS.Web/Controllers/TempController.cs

```diff
@@ -139,5 +141,79 @@ public ActionResult RetestCompileTimeoutSubmissionsFromCSharpDbAdvancedExam10Dec
 
             return this.Content($"Successfully enqueued {submissionIds.Count()} submissions for retesting.");
         }
+
+        public ActionResult MigrateProblemsToIndividualGroups()
+        {
+            var contests = this.Data.Contests
+                .AllWithDeleted()
+                .Include(c => c.Problems)
+                .Where(c => c.NumberOfProblemGroups <= 1 && !c.ProblemGroups.Any())
+                .ToList();
+
+            var output = new StringBuilder();
+            output.Append($"Done! contests affected: {contests.Count} <ol>");
+
+            foreach (var contest in contests)
+            {
+                foreach (var problem in contest.Problems)
+                {
+                    var problemGroup = new ProblemGroup
+                    {
+                        OrderBy = problem.OrderBy,
+                        CreatedOn = DateTime.Now
+                    };
+
+                    problemGroup.Problems.Add(problem);
+
+                    contest.ProblemGroups.Add(problemGroup);
+                }
+
+                output.Append($"<li>For Contest with <strong>ID:{contest.Id}</strong>: " +
+                    $"Added <strong>{contest.Problems.Count}</strong> Problem Groups</li>");
+            }
+
+            this.Data.SaveChanges();
+
+            output.Append("</ol>");
+            return this.Content(output.ToString());
+        }
+
+        public ActionResult MigrateProblemsToSharedGroups()
+        {
+            var contests = this.Data.Contests
+                .AllWithDeleted()
+                .Include(c => c.Problems)
+                .Where(c => c.NumberOfProblemGroups > 1 && !c.ProblemGroups.Any())
+                .ToList();
+
+            var output = new StringBuilder();
+            output.Append($"Done! contests affected: {contests.Count} <ol>");
+
+            foreach (var contest in contests)
+            {
+                output.Append($"<li>Created Problem Groups for Contest: {contest.Name} (ID:{contest.Id})<ol>");
+                for (var i = 0; i < contest.NumberOfProblemGroups; i++)
```

Comment:
`NumberOfProblemGroups`  looks like redundant after introducing ProblemGroups, right?

Reply:
For this issue we are keeping this property, will be refactored further more as part of the bigger issue for this problem https://github.com/SoftUni-Internal/suls-issues/issues/3777

Reply:
:shipit: